File: FrontEnd IMMRequest/src/app/models/Area.ts
import { Tema } from './Tema';
export class Area {
    id : Int32Array;
    nombre: string;
    temas :Array<Tema>;
   
  
    constructor(id:Int32Array, nombre:string,temas :Array<Tema>){
        this.nombre = nombre;
        this.id = id;
        this.temas = temas;
    }
  }

File: BackEnd IMMRequest/IMMRequest.Domain/Area.cs
using System;
using System.Collections.Generic;

namespace IMMRequest.Domain
{
    public class Area
    {
        public Area()
        {
        }

        public Area(string nombre)
        {
            Nombre = nombre;
        }

        public int Id{get ; set; }
        public String Nombre{get ; set; }

        public List<Tema> Temas{get ; set; }

        public void seteartemas(List<Tema> temas){
            Temas= temas;
        }

         public void agregartema(Tema tema){
            if(tema.Nombre!=""){
                    if (!Temas.Exists(x=> x.Nombre==tema.Nombre)){
                        Temas.Add(tema);
                        
                    }else{
                        throw new ArgumentException("Ya existe un Tema con ese nombre");
                    }
                }else{
                        throw new ArgumentException("El Nombre del Tema no puede ser vacio");
                }
        }
    }
}
